ingest/state.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+)
+
+type FileStatus string
+
+const (
+	FileStatusProcessed FileStatus = "processed"
+	FileStatusFailed    FileStatus = "failed"
+)
+
+type FileStateEntry struct {
+	Filename  string     `json:"filename"`
+	Status    FileStatus `json:"status"`
+	Timestamp time.Time  `json:"timestamp"`
+	Error     string     `json:"error,omitempty"`
+}
+
+type StateManager struct {
+	stateFilePath string
+	mu            sync.RWMutex
+	state         map[string]FileStateEntry
+	logger        *IngestLogger
+}
+
+func NewStateManager(stateFilePath string, logger *IngestLogger) (*StateManager, error) {
+	sm := &StateManager{
+		stateFilePath: stateFilePath,
```

Comment:
I suppose for now it makes sense to put this kind of state in a file. But we should probably consider where we want to keep this kind of thing in the future, assuming we'll want to share state between multiple hosts that are doing this kind of processing. I'll make a task for it so we remember to discuss it later.